### PR TITLE
feat(indexer): NEAR Indexer can init configs

### PR DESCRIPTION
**Pull request by khorolets**
* Expose `neard::init_configs` into NEAR Indexer. 
* Update `tools/indexer/example` and NEAR Indexer README according to changes.

### CODE REVIEW

File: tools/indexer/example/src/main.rs [hunk 1]
```diff
@@ -3,12 +3,61 @@ use std::io;
...
+#[derive(Clap, Debug)]
+enum SubCommand {
+    /// Run NEAR Indexer Example. Start observe the network
+    Run,
+    /// Initialize necessary configs
+    Init(ConfigArgs),
+}
+
+#[derive(Clap, Debug)]
+struct ConfigArgs {
```
Comment (frol) [inline tools/indexer/example/src/main.rs:34]: Let's call it `InitConfigArgs`

File: tools/indexer/example/src/main.rs [hunk 2]
```diff
@@ -3,12 +3,61 @@ use std::io;
...
 use near_indexer;
 
+/// NEAR Indexer Example
+/// Watches for stream of blocks from the chain
+#[derive(Clap, Debug)]
+#[clap(version = "0.1", author = " Near Inc. <hello@nearprotocol.com>")]
+struct Opts {
+    /// Sets a custom config dir. Defaults to ~/.near/
+    #[clap(short, long, default_value = "~/.near/")]
+    home_dir: String,
```
Comment (frol) [inline tools/indexer/example/src/main.rs:20]: Can we make it PathBuf to be explicit with our requirements?

File: chain/indexer/src/lib.rs [hunk 3]
```diff
@@ -50,6 +49,35 @@ impl Indexer {
...
+        test_seed: Option<&str>,
+        num_shards: near_primitives::types::NumShards,
+        fast: bool,
+        genesis: Option<&str>,
+        download: bool,
+        download_genesis_url: Option<&str>,
+    ) {
+        if download {
+            openssl_probe::init_ssl_cert_env_vars();
+        }
```
Comment (frol) [inline chain/indexer/src/lib.rs:67]: I believe we need this fix for `neard init` as well, so there is no point in using this wrapper. Can you check `neard init` on your laptop?

File: chain/indexer/Cargo.toml [hunk 4]
```diff
@@ -8,6 +8,7 @@ edition = "2018"
 actix = "0.9"
 tracing = "0.1.13"
 futures = "0.3.5"
+openssl-probe = { git = "https://github.com/khorolets/openssl-probe.git", branch="feature/support-macports" }
```
Comment (frol) [inline chain/indexer/Cargo.toml:11]: 1. I think we should depend on a regular package and patch it (define a patch section).
2. I believe we should move the initialization of `openssl_probe` to `neard` (move from telemetry), so telemetry and genesis downloads work.

File: Cargo.toml [hunk 5]
```diff
@@ -47,6 +47,10 @@ members = [
 [dependencies]
 # pins this version to fix https://github.com/actix/actix-net/issues/129
 actix-rt = "=1.1.1"
+openssl-probe = { version = "0.1.2"}
+
+[patch.crates-io]
+openssl-probe = { git = "https://github.com/khorolets/openssl-probe.git", branch="feature/support-macports" }
```
Comment (bowenwang1996) [inline Cargo.toml:53]: is this supposed to be a long term solution?

File: chain/indexer/src/lib.rs [hunk 6]
```diff
@@ -50,6 +49,32 @@ impl Indexer {
         Self { actix_runtime: system, view_client, client, near_config }
     }
 
+    /// Initializes genesis and client configs and stores in the given folder
+    /// Exposes `neard::init_configs()`
+    pub fn init_configs(
```
Comment (bowenwang1996) [inline chain/indexer/src/lib.rs:54]: This looks suboptimal. Why do you need to do this?

File: tools/indexer/example/src/main.rs [hunk 7]
```diff
@@ -215,11 +264,30 @@ async fn listen_blocks(mut stream: mpsc::Receiver<near_indexer::BlockResponse>)
 }
 
 fn main() {
-    let home_dir: Option<String> = env::args().nth(1);
-
+    openssl_probe::init_ssl_cert_env_vars();
```
Comment (frol) [inline tools/indexer/example/src/main.rs:267]: Let's add a comment that we use it to automatically search the for root certificates to perform HTTPS calls (sending telemetry and downloading genesis)

File: tools/indexer/example/src/main.rs [hunk 8]
```diff
@@ -3,12 +3,61 @@ use std::io;
...
 use tokio::sync::mpsc;
 use tracing_subscriber::filter::LevelFilter;
 use tracing_subscriber::EnvFilter;
 
 use near_indexer;
 
+/// NEAR Indexer Example
+/// Watches for stream of blocks from the chain
+#[derive(Clap, Debug)]
+#[clap(version = "0.1", author = " Near Inc. <hello@nearprotocol.com>")]
```
Comment (frol) [inline tools/indexer/example/src/main.rs:16]:
```suggestion
#[clap(version = "0.1", author = "Near Inc. <hello@nearprotocol.com>")]
```

File: Cargo.toml (hunk 5 above)
Comment (frol) [inline Cargo.toml:53]: Add a comment on why we patch and what is the condition to drop the patch (reference the relevant conversation on the upstream)

Comment (khorolets) [reply Cargo.toml:53]: It is supposed to be removed once author of `openssl-probe` publish 0.2.0 version with fix, or fix the breaking change for 0.1.3

File: chain/indexer/src/lib.rs (hunk 6 above)
Comment (khorolets) [reply chain/indexer/src/lib.rs:54]: Honestly, I have no idea, I'm already fixing it :)

File: tools/indexer/example/src/main.rs [hunk 9]
```diff
@@ -3,12 +3,61 @@ use std::io;
...
 
 use near_indexer;
 
+/// NEAR Indexer Example
+/// Watches for stream of blocks from the chain
+#[derive(Clap, Debug)]
+#[clap(version = "0.1", author = "Near Inc. <hello@nearprotocol.com>")]
+struct Opts {
+    /// Sets a custom config dir. Defaults to ~/.near/
+    #[clap(short, long, default_value = "~/.near/")]
```
Comment (frol) [inline tools/indexer/example/src/main.rs:19]: I wonder if that actually works as `~` is only resolved to the home folder on the shell level but if you explicitly try to open/create a folder `"~/.near"` is will treat it like a relative path with `~` being a literal naming of a folder, not the home folder.

Comment (khorolets) [reply tools/indexer/example/src/main.rs]: good catch, fixed

**Comment by khorolets** [comment]
> Does it make sense to move indexer out of the `chain` directory?

@bowenwang1996 Do you have a suggestion of better place?

**Comment by frol** [comment]
It makes sense to move it out of nearcore repo, but that will require some setup, and, ideally, have nearcore crates published on crates.io. So far we just put it there to shortcut the maintenance costs.

**Comment by khorolets** [comment]
@bowenwang1996 @frol so can we merge it?

**Comment by frol** [comment]
Yes, pull the trigger

**Pull request merged**